### PR TITLE
Different AreaStyle for each Y-value for StairstepPlot, CategoryAxisModel's firstCategoryIsZero.

### PR DESCRIPTION
**Pull request by KentVu**
I've customized `StairstepPlot` further to allow different `AreaStyle` for each Y-value, along with customizable `lineStyle` for each Y-value "cap", as shown below:
Styled StairstepPlot
_(The "cap" and the area can have the same color, or all the areas can have the same color, if one wishes)._

In order for the area to fill to the X-axis (y = 0), I have to modify `CategoryAxisModel` to place first _category_ at the zero position, if the `firstCategoryIsZero` parameter is set to `true`.

### CODE REVIEW

**Comment by KentVu** [comment]
_(Refer to my last PR (#35) for usage example)_

File: src/commonMain/kotlin/io/github/koalaplot/core/line/LinePlot.kt [hunk 1]
```diff
@@ -514,6 +512,30 @@ private fun <X, Y> XYGraphScope<X, Y>.generateArea(
     }
 }
 
+private fun DrawScope.fillRectangle(
```
Comment (gsteckman) [inline src/commonMain/kotlin/io/github/koalaplot/core/line/LinePlot.kt:515]: Does this draw the rectangle differently than DrawScope.drawRect? Why couldn't that be used?

**Review by gsteckman — COMMENTED** [review]
Did you check if DrawScope.drawRect could be used instead of writing a custom version?

**Comment by KentVu** [comment]
@gsteckman 
Thank you for your suggestion, I didn't aware of that API.
I've changed to using `DrawScope.drawRect`, so please kindly check.

**Review by gsteckman — APPROVED** [review]

**Comment by gsteckman** [comment]
Thanks for your contribution!

**Pull request merged**